app/controllers/billing_expenses_controller.rb

```diff
@@ -52,6 +45,34 @@ def set_volunteer
     @volunteer = Volunteer.find(params[:volunteer_id]) if params[:volunteer_id]
   end
 
+  def set_volunteer_hours
+    @volunteer_hours = @volunteer.hours.where(billing_expense: nil)
+  end
+
+  def compute_hours
+    return if @volunteer_hours.empty?
+    hours = @volunteer_hours.sum(:hours)
+    minutes = @volunteer_hours.sum(:minutes)
```

Comment:
This runs two `SUM(...)` queries, you can do `sum(&:hours)` / `sum(&:minutes)` instead to load the objects only once and sum the values in Ruby instead of SQL.